src/main/java/com/teragrep/lsh_01/HttpServerHandler.java

```diff
@@ -59,6 +63,11 @@ public HttpServerHandler(
 
     @Override
     public void channelRead0(ChannelHandlerContext ctx, FullHttpRequest msg) {
+        if (!ctx.channel().isActive()) {
+            LOGGER.error("The channel bound to the ChannelHandlerContext is not connected, can't handle the message.");
+            return;
```

Comment:
channel does not need to be active anymore in order to process the received FullHttpRequest. Please try/catch the remote address or find an alternative way to capture it.

Reply:
Decided to use the try/catch, is the new change ok?

Reply:
please try to extract the remote address from FullHttpRequest or ctx instead using the channel() if possible. Skipping a message from a disconnected client is not tolerable.